src/formatDate.js

```diff
@@ -50,7 +50,69 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const splittedDate = date.split(fromSeparator);
+  const formattedDate = [];
+  let currentYear = 'Y';
+  let currentMonth = 'M';
+  let currentDay = 'D';
+
+  for (let i = 0; i < fromFormat.length - 1; i += 1) {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        currentYear += splittedDate[i];
+        break;
+
+      case 'M':
+        currentMonth += splittedDate[i];
+        break;
+
+      case 'D':
+        currentDay += splittedDate[i];
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i += 1) {
+    let year = currentYear.slice(1);
+    const month = currentMonth.slice(1);
+    const day = currentDay.slice(1);
+
+    if (year.length > toFormat[i].length) {
+      year = year.slice(2);
+    }
+
+    if (year.length < toFormat[i].length && +year < 30) {
+      year = '20' + year;
+    }
+
+    if (year.length < toFormat[i].length && +year >= 30) {
+      year = '19' + year;
+    }
+
+    switch (toFormat[i][0]) {
+      case currentYear[0]:
```

Comment:
just use `Y`, `M`, `D` or create the constants for them (e.g. `const YEAR_DIGIT = 'Y'`)